### PR TITLE
unitest for iip15-manager contract

### PR DESCRIPTION
**Pull request by millken**
# Description
tests case for iip15manager contract

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by dustinxie — APPROVED** [review]

File: action/protocol/execution/testdata-istanbul/iip15-manager.sol [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    function getRecipient(address _contractAddress) public view returns (address) {
+        return contractMapping[_contractAddress];
+    }
+
+    function isContractApproved(address _contractAddress) public view returns (bool) {
+        return isApproved[_contractAddress];
+    }
+
+}
```
Comment (Liuhaai) [inline action/protocol/execution/testdata-istanbul/iip15-manager.sol:43]: new line

File: action/protocol/execution/testdata-istanbul/iip15-manager.json [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        "rawAccessList": [],
+        "rawExpectedGasConsumed": 38216,
+        "expectedStatus": 1,
+        "expectedErrorMsg": "",
+        "expectedLogs": [{}],
+        "rawReturnValue": "",
+        "comment": "approveContract use new owner, success"
+    }
+]
+}
```
Comment (Liuhaai) [inline action/protocol/execution/testdata-istanbul/iip15-manager.json:254]: same

File: action/protocol/execution/testdata-istanbul/iip15-manager.sol [hunk 3]
```diff
@@ -0,0 +1,43 @@
+//SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.0;
+import "https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/access/Ownable.sol";
+
+contract IIP15Manager is Ownable {
+    mapping (address => address) private contractMapping;
+    mapping (address => bool) private isApproved;
```
Comment (ququzone) [inline action/protocol/execution/testdata-istanbul/iip15-manager.sol:8]: Maybe merge contractMapping and isApproved into struct is prefect.

**Comment by ququzone** [comment]
By the way, does it need `deregister` function? If want to add extra APIs in the feature then need to deploy this contract as upgradeable.

**Comment by millken** [comment]
@ququzone @dustinxie update contract, use map => struct to store contract info, added `getContract`, `removeContract` APIs, deleted `isApproved`,`isRegistered` API(use getContract instead of, only call once)

**Review by ququzone — APPROVED** [review]

File: action/protocol/execution/testdata-istanbul/iip15-manager.sol [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+pragma solidity ^0.8.0;
+
+import "https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/access/Ownable.sol";
+
+contract IIP15Manager is Ownable {
+
+    struct ContractInfo {
+        address recipient;
+        bool isRegistered;
```
Comment (ququzone) [inline action/protocol/execution/testdata-istanbul/iip15-manager.sol:11]: This field is unnecessary and you can use recipient is address(0) to check whether registered.

**Pull request merged**